### PR TITLE
Add an API for variable font axes in TextStyle and connect it to SkParagraph

### PR DESCRIPTION
**Pull request by jason-simmons**
See https://github.com/flutter/flutter/issues/87291

### CODE REVIEW

File: lib/ui/text.dart [hunk 1]
```diff
@@ -933,6 +933,64 @@ class FontFeature {
...
+      assert(value != null);
+
+  /// The tag that identifies the design axis.  Must consist of 4 ASCII
+  /// characters.
+  final String axis;
+
+  /// The value assigned to this design axis.
+  ///
+  /// The range of usable values depends on the specification of the axis.
+  final double value;
```
Comment (dnfield) [inline lib/ui/text.dart:966]: Does it make sense to document these? Are there commonly used values? Can we document more about the acceptable ranges of each character?

**Review by dnfield — COMMENTED** [review]
Code and test lgtm. An example would help in the docs

File: lib/ui/text.dart [hunk 2]
```diff
@@ -933,6 +933,64 @@ class FontFeature {
   String toString() => "FontFeature('$feature', $value)";
 }
 
+/// An axis tag and value that can be used to customize variable fonts.
+///
+/// Some fonts are variable fonts that can generate a range of different
+/// font faces by altering the values of the font's design axes.
+///
+/// See https://docs.microsoft.com/en-us/typography/opentype/spec/otvaroverview
```
[the remark this answers is not in the document]
Comment (jason-simmons) [reply lib/ui/text.dart:941]: done

File: lib/ui/text.dart (hunk 1 above)
Comment (jason-simmons) [reply lib/ui/text.dart:969]: The usable values depend on the definition of each axis.  The user will need to consult the documentation for the font and the font format (such as the OpenType spec linked above)

File: lib/ui/text.dart [hunk 3]
```diff
@@ -933,6 +933,67 @@ class FontFeature {
...
+///
+/// Example:
+/// `TextStyle(fontVariations: <FontVariation>[FontVariation('wght', 800.0)])`
```
Comment (dnfield) [inline lib/ui/text.dart:944]: nit: consider guarding this with ``` and some kind of comment explaining about what this would do with the Roboto font.

**Review by dnfield — APPROVED** [review]
LGTM

**Comment by dnfield** [comment]
Analysis unhappy

**Pull request merged**